src/utils/jester_etl/jester.py

```diff
@@ -0,0 +1,191 @@
+
+    Returns:
+        list: A list of strings of JSON objects.
+    """
+    output = []
+    for i, val in enumerate(row):
+        # The first element is the number of non-zero elements, which is
+        # useless to us
+        if i == 0:
+            continue
+        # 99 is used as the NULL character
+        if val == 99:
+            continue
+        # Otherwise element i corresponds to the rating for joke i
+        current_rating = deepcopy(RATINGS)
+        current_rating["user_id"] = user_id
+        current_rating["joke_id"] = i
+        current_rating["rating"] = val
+        output.append(json.dumps(current_rating))
+
+    return output
+
+
+def block_generator(filename):
+    """Generates a set of "data blocks" from the joke file.
+
+    Each block corresponds to the data from a single joke, with the following
+    form:
+
+    block = [
+        "Int:", # Joke Number
+        "HTML joke",
+        ...
+        "End of HTML joke."
+    ]
+
+    Args:
+        filename (str): The name of the file to load.
+
+    Yields:
+        list: A list of all the lines in a file relating to a single joke, in
+            order.
+
+    """
+    buffer = []
+    with open(filename) as f:
+        for line in f:
+            line = line.strip()
+            if not line:
+                yield buffer
+                buffer = []
+            else:
+                buffer.append(line)
+
+
+def joke_to_json(path):
+    """Convert the joke file to a list of JSON objects.
+
+    Args:
+        path (str): The location of the joke file.
+
+    Returns:
+        list: A list of JSON objects represented as strings.
+
+    """
+
+    output = []
+    for block in block_generator(path):
+        current_joke = deepcopy(JOKES)
+
+        # The first line is the number, with a colon at the end
+        current_joke["joke_id"] = int(block[0][:-1])
+
+        # All remaining lines are the text of the joke as HTML
+        text = '\n'.join(block[1:])
+        current_joke["joke_text"] = bs4.BeautifulSoup(text, "lxml").get_text().strip()
+
+        output.append(json.dumps(current_joke))
+
+    return output
+
+
+if __name__ == "__main__":
+
+    import argparse
+
+    # Set up command line flag handling
+    parser = argparse.ArgumentParser(
+            description="Transform the Jester datasets to JSON",
+        )
+    parser.add_argument(
+            'joke_file',
+            type=str,
+            help="the file containing the jokes, normally 'jester_items.dat'",
+            )
+    parser.add_argument(
+            'ratings_file',
+            type=str,
+            nargs='+',
+            help="the xls files containing the ratings",
+            )
+    parser.add_argument(
+            '-o',
+            '--output_directory',
+            type=str,
+            action="store",
+            help="the directory to save the output JSON files, by default the current directory",
+            default="./",
+            )
+
```

Comment:
If the directory does not exist, it will exit without creating the output files. We can add this option in the future.